ccw.core/src/java/ccw/CCWPlugin.java

```diff
@@ -109,31 +105,18 @@ public static ITracer getTracer() {
 			return NullTracer.INSTANCE;
 	}
     
-    public synchronized void startREPLServer() {
-    	if (ackREPLServer == null) {
-	        try {
-	        	// TODO use ClojureOSGi.withBundle instead
-	        	Var startServer = BundleUtils.requireAndGetVar(getBundle().getSymbolicName(), "clojure.tools.nrepl.server/start-server");
-	        	Object defaultHandler = BundleUtils.requireAndGetVar(
-	        	        getBundle().getSymbolicName(),
-	        	        "clojure.tools.nrepl.server/default-handler").invoke();
-	        	Object handler = BundleUtils.requireAndGetVar(
-	        	        getBundle().getSymbolicName(),
-	        	        "clojure.tools.nrepl.ack/handle-ack").invoke(defaultHandler);
-	            ackREPLServer = (ServerSocket)((Map)startServer.invoke(Keyword.intern("handler"), handler)).get(Keyword.intern("server-socket"));
-	            CCWPlugin.log("Started ccw nREPL server: nrepl://127.0.0.1:" + ackREPLServer.getLocalPort());
-	        } catch (Exception e) {
-	            CCWPlugin.logError("Could not start plugin-hosted REPL server", e);
-	            throw new RuntimeException("Could not start plugin-hosted REPL server", e);
-	        }
+    public void startREPLServer() {
+    	try {
+    	    ClojureInvoker.newInvoker(this, "ccw.core.launch")._("ccw-nrepl-start-if-necessary");
+    	} catch (Exception e) {
+    		CCWPlugin.logError("Could not start plugin-hosted REPL server", e);
+    		throw new RuntimeException("Could not start plugin-hosted REPL server", e);
     	}
     }
     
     public int getREPLServerPort() {
-    	if (ackREPLServer == null) {
-    		startREPLServer();
-    	}
-    	return ackREPLServer.getLocalPort();
+    	startREPLServer();
+    	return (Integer) ClojureInvoker.newInvoker(this, "ccw.core.launch")._("ccw-nrepl-port");
```

Comment:
Hmm, I hadn't understood that newInvoker would be called again each time getREPLServerPort() is called. I had thought it would be done one per class. Let's keep this in mind if people report that things are slower.